resources/assets/js/features/search/components/Result/_Modals/SubscribeModal.js

```diff
@@ -0,0 +1,268 @@
+            animals: null,
+            luggage: null,
+            seats: null,
+            rating: null,
+            requestSendSuccess: false
+        };
+        this.animalsChange = this.animalsChange.bind(this);
+        this.luggageChange = this.luggageChange.bind(this);
+        this.seatsChange = this.seatsChange.bind(this);
+        this.ratingChange = this.ratingChange.bind(this);
+    }
+
+    updateFilterData(data) {
+        this.setState({
+            time: data.filters.time,
+            animals: data.filters.animals,
+            luggage: data.filters.luggage,
+            seats: data.filters.seats,
+            rating: data.filters.rating,
+            start_at: data.start_at
+        });
+    }
+
+    componentWillMount() {
+        this.updateFilterData(this.props.data);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.updateFilterData(nextProps.data);
+    }
+
+    animalsChange(e) {
+        this.setState({animals: e.target.value});
+    }
+
+    luggageChange(e) {
+        this.setState({luggage: e.target.value});
+    }
+
+    seatsChange(e) {
+        this.setState({seats: e.target.value});
+    }
+
+    ratingChange(e) {
+        this.setState({rating: e.target.value});
+    }
+
+    onSubmit(e) {
+        e.preventDefault();
+        const formEmail = e.target['email'].value,
+            {email, data} = this.props,
+            {time, animals, luggage, seats, rating, price} = this.state;
+
+        let subsEmail = formEmail ? formEmail : email;
+        let toBeTransformed = {
+            time,
+            animals,
+            luggage,
+            seats,
+            rating,
+            price,
+            subsEmail,
+            data
+        };
+        let requestData = transformSubscriptionData(toBeTransformed);
+        simpleRequest.post('api/v1/subscription', requestData)
+            .then((response) => {
+                if (response.status === 200) {
+                    this.setState({requestSendSuccess: true});
+                }
+            });
+    }
+
+    render() {
+        const {translate, isOpen, onClosed, data, isAuth, email} = this.props;
+        const {time, price, animals, seats, luggage, rating, requestSendSuccess} = this.state;
+        const authClass = (isAuth && email) ? ' subscribe-modal__footer-label-hidden' : '';
+
+        return (
+            <span>
+
+                <Modal className="subscribe-modal" size="lg" isOpen={isOpen} toggle={onClosed}>
+                    <ModalHeader className="subscribe-modal__header" toggle={onClosed}>
+                        <i className="subscribe-modal-icon fa fa-rss" aria-hidden="true" title={translate('subscription.info')}/>
+                        <span className="subscribe-modal__header-text" title={translate('subscription.info')}>{translate('subscription.subscribe_header')}</span>
+                    </ModalHeader>
+                    <ModalBody className="p-15">
+                    {(requestSendSuccess) ?
```

Comment:
specify method that will return needed html base on state, no need to render template parts in this way